rust/src/core.rs

```diff
@@ -163,6 +166,83 @@ impl FilesystemScriptPrep {
     }
 }
 
+/// Some Fedora/RHEL kernels ship .hmac files with absolute paths inside,
+/// which breaks when we relocate them into ostree/.  This function
+/// changes them to be relative.
+///
+/// Fixes are pending to ensure the kernels are doing this from the start,
+/// but we can't rely on that landing for a while. In addition, there are
+/// many 3rd-party kernels that have replicated the behavior in the ARK
+/// kernel.spec.
+fn verify_kernel_hmac_impl(moddir: &Dir) -> Result<()> {
+    use std::os::unix::ffi::OsStrExt;
+
+    // FIXME: in 2023
+    // This method is intentionally a misnomer because it should eventually
+    // be changed to the "sanity check" (below). It currently patches absolute
+    // paths to give kernel package maintainers time to update their .spec files.
+
+    const SEPARATOR: &[u8] = b"  ";
+
+    let hmac_path = ".vmlinuz.hmac";
+
+    let hmac_contents = if let Some(f) = moddir.open_optional(hmac_path)? {
+        let mut f = BufReader::new(f);
+        let mut s = Vec::new();
+        f.read_to_end(&mut s)?;
+        s
+    } else {
+        return Ok(());
+    };
+
+    if !hmac_contents.contains(&b'/') {
+        return Ok(());
+    }
+
+    let split_index = hmac_contents
+        .windows(SEPARATOR.len())
+        .position(|v| v == SEPARATOR)
+        .ok_or_else(|| {
+            anyhow::anyhow!(
+                "Missing path in .vmlinuz.hmac: {}",
+                String::from_utf8_lossy(hmac_contents.as_slice())
+            )
+        })?;
+
+    let (hmac, path) = hmac_contents.split_at(split_index + SEPARATOR.len());
```

Comment:
So I think this code would be more elegant using https://doc.rust-lang.org/std/primitive.str.html#method.split_once

I recently converted some code like this:
https://github.com/ostreedev/ostree-rs-ext/pull/187/commits/0ec9e980d3274a3a8fe2f99d3fd253c8d03fb160
https://github.com/ostreedev/ostree-rs-ext/pull/188/commits/88e14fee7b4fe940ce7dbe9e3daa0213192706b5